sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -723,6 +725,12 @@ def fit(self, X, y=None):
                 "Cannot store centers when using a precomputed distance matrix."
             )
 
+        if self.metric == "cosine" and self.algorithm not in ["brute", "auto"]:
+            raise ValueError(
+                "The 'cosine' metric is only supported with the 'brute' and 'auto'"
+                " algorithms."
+            )
+
```

Comment:
Maybe a warning instead or error, since the doc says:
> If the X passed during fit is sparse or metric is invalid for both KDTree and BallTree, then it resolves to use the "brute" algorithm.

Reply:
Actually we don't need to do anything. The behavior is documented and if the metric is only supported by brute, it will fallback to brute automatically

Reply:
I agree with @jeremiedbb. When a metric such as "cosine" is passed with an algorithm that doesn't support this metric, an error will be raised later in the fit function. This is already implemented.